### PR TITLE
Self-hosted analytics

### PR DESCRIPTION
**Pull request by finestructure**
New deployment variables:

- PLAUSIBLE_ADMIN_USER_EMAIL
- PLAUSIBLE_ADMIN_USER_NAME
- PLAUSIBLE_ADMIN_USER_PWD
- PLAUSIBLE_BASE_URL
- PLAUSIBLE_DOMAIN
- PLAUSIBLE_SECRET_KEY_BASE
- MAIL_RELAY_HOST
- MAIL_RELAY_PORT
- MAIL_RELAY_USERNAME
- MAIL_RELAY_PASSWORD

added to

- [x] dev
- [ ] prod

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -164,8 +171,52 @@ services:
     image: prom/pushgateway:v1.3.0
     restart: always
 
+  # Analytics
+
+  mail:
+    image: bytemark/smtp
+    restart: always
```
Comment (finestructure) [inline docker-compose.yml:178]: Unsurprisingly, this isn't working. We could perhaps live without email notifications from the service but the problem is that it gates the login:

CleanShot 2021-01-18 at 11 39 18@2x

Comment (finestructure) [reply docker-compose.yml:178]: Mail logs:

```
  241 delivering 1l1Rsi-00003s-Pf
  241 R: dnslookup for sas@finestructure.co
  242 T: remote_smtp for sas@finestructure.co
  242 Connecting to mx.hover.com.cust.hostedemail.com [216.40.42.4]:25 ... connected
  242   SMTP<< 450 4.7.1 Client host rejected: cannot find your reverse hostname, [208.52.185.184]
  242   SMTP>> QUIT
  242   SMTP(close)>>
  242 LOG: MAIN
  242   H=mx.hover.com.cust.hostedemail.com [216.40.42.4]: SMTP error from remote mail server after initial connection: 450 4.7.1 Client host rejected: cannot find your reverse hostname, [208.52.185.184]
  241 LOG: MAIN
  241   == sas@finestructure.co R=dnslookup T=remote_smtp defer (0) H=mx.hover.com.cust.hostedemail.com [216.40.42.4]: SMTP error from remote mail server after initial connection: 450 4.7.1 Client host rejected: cannot find your reverse hostname, [208.52.185.184]
  243 LOG: MAIN
  243   <= hello@plausible.local H=spi_plausible_1.spi_default (spi-dev.swiftpackageindex.com) [172.19.0.18] P=esmtp S=1026
  243 LOG: smtp_connection MAIN
  243   SMTP connection from spi_plausible_1.spi_default (spi-dev.swiftpackageindex.com) [172.19.0.18] closed by QUIT
  244 Exim version 4.89 uid=101 gid=101 pid=244 D=40001
```

We'd need an authenticated SMTP that we can relay to and which allows us to connect from a container that doesn't support reverse IP lookup.

Comment (daveverwer) [reply docker-compose.yml:178]: There are a few providers that will give us a few emails/day for free on "developer" plans, which is all we'd ever need for this.

https://www.mailjet.com/pricing/ give us 200/day going to ~$10/mo after 6,000 emails/mo.
https://sendgrid.com/pricing/ give 100/day going ~$15/mo after 3,000 emails/mo

Both are reputable companies that maintain good sending reputation.

It's also possible that we'll want to expand this as we look at email functionality in the future, although we'd need to approach this carefully as email sending can get expensive quickly.

Should we configure a Mailjet account?

Comment (daveverwer) [reply docker-compose.yml:178]: https://www.sendinblue.com/pricing/ do 300/day and seem reputable from a bit of research, but I've not used them before. I've used both Mailjet and Sendgrid in previous projects.

Comment (finestructure) [reply docker-compose.yml:178]: Same, but I think we can use a gmail account for sending. Trying that out right now!

**Comment by finestructure** [comment]
This is a problem I'm not sure we can solve without also setting up a reverse proxy or getting a Cloudflare Pro account:

CleanShot 2021-01-18 at 15 13 56@2x

**Comment by finestructure** [comment]
I'm guessing it's Cloudflare that's rewriting the `http` to `https`. Unless we can create an exception I don't see how we could work around this.

**Comment by finestructure** [comment]
Yeah, this won't work without a reverse proxy or Cloudflare routing to services on ports:

CleanShot 2021-01-19 at 07 18 58@2x

**Comment by finestructure** [comment]
Closing for now, as discussed, and moving branch to attic.

**Pull request closed without merging**